src/engine/input/RunOnKeyAttribute.cs

```diff
@@ -52,22 +57,25 @@ public bool ReadHeldOrPrimedAndResetPrimed()
 
     public override bool OnInput(InputEvent @event)
     {
-        bool wasOurKey = false;
+        bool result = false;
 
         if (@event.IsActionPressed(InputName))
         {
-            wasOurKey = true;
+            if (CallMethodInOnInput)
+                result = CallMethod(0.0f);
```

Comment:
This makes it so that it doesn't receive the delta time. So I'd like a property that the user can use to turn this off if they need a RunOnKey to get the passed time.

Reply:
do you mean what I implemented in the latest commit?

Reply:
https://github.com/Revolutionary-Games/Thrive/pull/1949/commits/db291d814c8ec0ca19835e697e5749e4c4800f69

Reply:
No, I meant adding one more property. Something like CallbackNeedsElapsedTime, that if it is true then overrides `CallMethodInOnInput` and blocks `CallMethod(0.0f);`

Reply:
yeah, but you can set `CallMethodWithZeroOnInput` itself on RunOnKeyAttribute, which prohibits calling the method with 0.0f

Reply:
uhh, okay, I can kinda accept that, but it is bad design that the child classes don't let you set that attribute. It's a good code principle that an operation that is in the base class also works on child classes. So...
I think that renaming CallMethodWithZeroOnInput back to what it was before, would be good. And then add a new property "CallbackRequiresElapsedTime" that defaults to true (I think that's the more common case). Then the inputs that need to not consume things, can specify CallbackRequiresElapsedTime = false, and return false in the callback.

The code in here would then be `if (CallMethodInOnInput && !CallbackRequiresElapsedTime)` (and if that condition is true, the _Process call should then have a reverse check, to not call the callback twice, or should it work that way?)

Reply:
The _Process may call the callback twice, because the only logic where _Process gets called anyways is in the `RunOnKeyAttribute` class, which is designed to repeatedly call the callback method.